tests/utilities/test_auto_restart.py

```diff
@@ -917,22 +922,27 @@ def test_dataset_rng_states_restart_with_lightning(tmpdir, dataset_classes, mult
         multiple_trainloader_mode=multiple_trainloader_mode,
     )
 
-    all_batches, weights0 = _run_training(trainer_kwargs, dataset_classes)
-    all_batches = torch.stack(all_batches)
-    assert len(all_batches) == 9
+    # this test will fail `fault_tolerant` don't support multiple datasets.
+    # this tests works as the dataset is fully deterministic and therefore
+    # there is not overall between the seeds.
+    with mock.patch("pytorch_lightning.trainer.data_loading._validate_fault_tolerant_automatic", lambda x, y: None):
```

Comment:
mind adding this mock to the definition? this way we don't get the extra indentation

Reply:
Done !